### ISSUE TITLE
Correct license?

### ISSUE THREAD

**Issue by notlee** [skratchdot/color-blind#3]
Hey thanks for `skratchdot/color-blind`, looks like a super useful package! 
I'd love to use it in the project I'm working on but have a question about the 
license.

The comment for the Color Blindness Simulation function  mentions 
non-commercial use without HCIRN permission. As the MIT license allows 
commercial uses I think it's incompatible? 

Although I found on Matthew Wickline's site this update:
>20151129 UPDATE
>    HCIRN appears to no longer exist. This makes it impractical
>    for users to obtain permission from HCIRN in order to use
>    color_blind_sims() for commercial works. Instead:
>
>    This work is licensed under a
>    Creative Commons Attribution-ShareAlike 4.0 International License.
>    http://creativecommons.org/licenses/by-sa/4.0/

Source: http://colorlab.wickline.org/colorblind/colorlab/engine.js

CC4-BY-SA does allow commercial use, but requires anything which 
builds on the project to be distributed with the same license, so I think
this project should also be CC4-BY-SA or a [compatible license](https://creativecommons.org/share-your-work/licensing-considerations/compatible-licenses)?
https://creativecommons.org/licenses/by-sa/4.0/

**Comment by heberleh** [comment]
That is a great question. It feels like every colorblind-package is non-commercial because of some coefficients or calculations. How can that be? Were they patented or something?
I'm trying to contact people for this project https://galactic.ink/labs/index.php?dir=Color-Vision%2FJavascript%2F
but it is not even clear what is their license and perhaps, how does it connect to potentially 'missed' license, like from  HCIRN.

**Comment by Ennoriel** [comment]
Hello @heberleh !

Did you get an answer from HCIRN or one of the original author?

Thanks for the reply ;)

**Comment by heberleh** [comment]
No, nothing new about this. I think there is no problem... But let us know if you find out.

**Comment by skratchdot** [comment]
yes- i think this project was setup with the incorrect license from the start. i had a script that was adding mit to my source files.

it's very stupid to have both:
https://github.com/skratchdot/color-blind/blob/20dd5e8a6e615a88b1b0c91633e6ebd31c13b91d/lib/blind.js#L16-L22

and:
https://github.com/skratchdot/color-blind/blob/20dd5e8a6e615a88b1b0c91633e6ebd31c13b91d/lib/blind.js#L12

in the same file.  it looks like i didn't know how to properly dual license or bubble up the license of the original source file i was using.

based on https://opensource.stackexchange.com/a/7436 it seems like i need to convert this entire project to CC-BY-SA and i don't know how that affects people who have already been assuming it's MIT.

any suggestions on how to fix this?

**Comment by skratchdot** [comment]
I updated the license array in package.json and published v0.1.3:
https://www.npmjs.com/package/color-blind/v/0.1.3

Here's what I changed today:
https://github.com/skratchdot/color-blind/compare/0.1.1...0.1.3

**Comment by Ennoriel** [comment]
I think that noone should assume that because a library X is writen with the licence Y means that it does not inherit from one of its dependency a licence more restrictive. We always do... I only saw it because I checked the source code.

The licence Y should not have been set at first but everyone makes mistakes 😄

Thanks for updating the licence 😀

**Comment by Giwayume** [comment]
> It feels like every colorblind-package is non-commercial because of some coefficients or calculations. How can that be? Were they patented or something? 

If the formula was patented, that patent is now expired, as patents last a max of 20 years and the copyright states the code was created in 2000.

Copyright is a different thing. You can technically use their methodology since no patent could possibly be valid for it, but any code they wrote is copyrighted so you cannot copy/paste this "color_blind_sims" function verbatim without obeying their license. You could write your own version of the function that does the same thing, as stated, the technique cannot be patented.
